scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -258,14 +258,26 @@ private[spel] class Typer(
         }
 
       case e: ConstructorReference =>
+        // TODO: validate constructor parameters...
         withTypedChildren { _ =>
-          val className  = e.getChild(0).toStringAST
-          val classToUse = Try(evaluationContext.getTypeLocator.findType(className)).toOption
-          // TODO: validate constructor parameters...
-          val clazz = classToUse.flatMap(kl => classDefinitionSet.get(kl).map(_.clazzName))
-          clazz match {
-            case Some(typedClass) => valid(typedClass)
-            case None             => invalid(ConstructionOfUnknown(classToUse))
+          val className    = e.getChild(0).toStringAST
+          val classToUse   = Try(evaluationContext.getTypeLocator.findType(className)).toOption
+          val typingResult = classToUse.flatMap(kl => classDefinitionSet.get(kl).map(_.clazzName))
+          typingResult match {
+            case Some(tc @ TypedClass(_, _)) =>
+              val dimensionsField = e.getClass.getDeclaredField("isArrayConstructor")
```

Comment:
Let's encapsulate this hacky code in some method :)

Reply:
Right :) Done :+1: